code401challenges/array-reverse/ArrayReverse.java

```diff
@@ -0,0 +1,27 @@
+public class ArrayReverse {
+    public static void main(String args[]) {
+        int[] array1 = {1, 3, 4, 6, 2, 5, 10, 9, 20};
+        int[] reverseArray = reverseArray(array1);
+        for (int el : reverseArray){
+            System.out.println(el);
+        }
+    }
+    public static int[] reverseArray(int[] array) {
+        int halfway;
+        if (array.length % 2 == 0) {
+            halfway = array.length/2;
+        } else {
+            halfway = (array.length + 1)/2;
+        }
+        for(int i = 0; i < halfway; i++){
+            if (i == halfway && array.length % 2 != 0){
```

Comment:
Since your loop only runs if `i < halfway`, this will never be true: you won't enter the loop if `i == halfway`. So you can remove this conditional from your code and it'll still work just as well.